src/components/graph/menus/network-modification-node-editor.jsx

```diff
@@ -764,31 +764,39 @@ const NetworkModificationNodeEditor = () => {
     };
 
     const doCreateModificationsElements = () => {
-        // studyPath contains [studyElement, parentDirElement, parentDirElement, ..., RootDirElement]
-        const STUDY_ELEMENT_INDEX = 0;
-        const PARENT_DIRECTORY_INDEX = 1;
+        // studyPath contains [RootDirectoryElement, directoryElement, ...,  directoryElement, studyElement]
         if (!studyPath || studyPath.length < 2) {
             snackError({
                 messageTxt: 'unknown study directory',
                 headerId: 'errCreateModificationsMsg',
             });
             return;
         }
-        const studyDirectoryUuid =
-            studyPath[PARENT_DIRECTORY_INDEX].elementUuid;
-        const studyDirName =
+
+        const studyIndex = studyPath.length - 1; // Should always be the last element
+        const parentDirectoryIndex = studyPath.length - 2; // Should always be the second to last element
+
+        const studyName = studyPath[studyIndex].elementName;
+
+        const studyDirectoryPath =
             '/' +
             studyPath
-                .slice(PARENT_DIRECTORY_INDEX)
-                .reverse()
+                .filter((r) => r.type === ElementType.DIRECTORY)
```

Comment:
In another place (https://github.com/gridsuite/gridstudy-app/blob/de6b1aee232bbc492c027999a56f38c297034711/src/components/study-container.jsx#L390C28-L390C51), we use slice as before, keep it for consistency ?
```suggestion
                .slice(0, studyPath.length - 1)
```